src/langchain_google_memorystore_redis/vector_store.py

```diff
@@ -0,0 +1,764 @@
+
+    @staticmethod
+    def init_index(client: redis.Redis, index_config: IndexConfig):
+        """
+        Initializes a named VectorStore index in Redis with specified configurations.
+        """
+        if not isinstance(index_config, HNSWConfig):
+            raise ValueError("index_config must be an instance of HNSWConfig")
+
+        # Preparing the command string to avoid long lines
+        command = (
+            f"FT.CREATE {index_config.name} ON HASH PREFIX 1 {RedisVectorStore.get_key_prefix(index_config.name)} "
+            f"SCHEMA {index_config.field_name} VECTOR {index_config.type} "
+            f"6 TYPE {index_config.data_type} DIM {index_config.vector_size} "
+            f"DISTANCE_METRIC {index_config.distance_metric}"
+        )
+
+        try:
+            client.execute_command(command)
+        except redis.exceptions.ResponseError as e:
+            if re.match(r"Redis module key \w+ already exists", str(e)):
+                logger.info("Index already exists, skipping creation.")
+            else:
+                raise
+
+        # TODO: When "Redis module key \w+ already exists" is caught, we should
+        # call FT.INFO to validate if the index properties match the arguments.
+        # If not, an exception should be thrown. This check is pending support
+        # for FT.INFO in the client library.
+
+    @staticmethod
+    def drop_index(client: redis.Redis, index_name: str, index_only: bool = True):
+        """
+        Drops an index from the Redis database. Optionally, it can also delete
+        the documents associated with the index.
+
+        Args:
+            client (Redis): The Redis client instance used to connect to the database.
+                This client provides the necessary commands to interact with the database.
+            index_name (str): The name of the index to be dropped. This name must exactly
+                match the name of the existing index in the Redis database.
+            index_only (bool, optional): A flag indicating whether to drop only the index
+                structure (True) or to also delete the documents associated with the index (False).
+                Defaults to True, implying that only the index will be deleted.
+
+        Raises:
+            redis.RedisError: If any Redis-specific error occurs during the operation. This
+                includes connection issues, authentication failures, or errors from executing
+                the command to drop the index. Callers should handle these exceptions to
+                manage error scenarios gracefully.
+        """
+        if index_only == False:
+            raise ValueError("Not supported")
+
+        command = f"FT.DROPINDEX {index_name} {'DD' if not index_only else ''}".strip()
+        client.execute_command(command)
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        batch_size: int = 1000,
+        **kwargs: Any,
```

Comment:
Let's pass in ids as optional rather than use kwargs

Reply:
done in the new PR